code/controllers/subsystem/minimaps.dm

```diff
@@ -0,0 +1,281 @@
+	if(flags & MINIMAP_FLAG_BRAVO)
+		LAZYADD(update_targets[MINIMAP_STRING_BRAVO], holder)
+	if(flags & MINIMAP_FLAG_CHARLIE)
+		LAZYADD(update_targets[MINIMAP_STRING_CHARLIE], holder)
+	if(flags & MINIMAP_FLAG_DELTA)
+		LAZYADD(update_targets[MINIMAP_STRING_DELTA], holder)
+	updators_by_datum[target] = holder
+	update_targets_unsorted += target
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/remove_updator)
+
+/**
+ * Removes a atom from the subsystems updating overlays
+ */
+/datum/controller/subsystem/minimaps/proc/remove_updator(atom/target)
+	SIGNAL_HANDLER
+	var/datum/minimap_updator/holder = updators_by_datum[target]
+	updators_by_datum -= target
+	LAZYREMOVE(update_targets[MINIMAP_STRING_XENO], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_MARINE], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_ALPHA], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_BRAVO], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_CHARLIE], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_DELTA], holder)
+	update_targets_unsorted -= target
+
+/**
+ * Holder datum for a zlevels data, concerning the overlays and the drawn level itself
+ * The individual image trackers have a raw and a normal list
+ * raw lists just store the images, while the normal ones are assoc list of [tracked_atom] = image
+ * the raw lists are to speed up the Fire() of the subsystem so we dont have to filter through
+ */
+/datum/hud_displays
+	///Actual icon of the drawn zlevel with all of it's atoms
+	var/icon/hud_image
+	var/list/xeno_images = list()
+	var/list/xeno_images_raw = list()
+	var/list/marine_images = list()
+	var/list/marine_images_raw = list()
+	var/list/alpha_images = list()
+	var/list/alpha_images_raw = list()
+	var/list/bravo_images = list()
+	var/list/bravo_images_raw = list()
+	var/list/charlie_images = list()
+	var/list/charlie_images_raw = list()
+	var/list/delta_images = list()
+	var/list/delta_images_raw = list()
+
+/**
+ * Holder datum to ease updating of atoms to update
+ */
+/datum/minimap_updator
+	/// Atom to update with the overlays
+	var/atom/minimap
+	///Target zlevel we want to be updating to
+	var/ztarget = 0
+
+/datum/minimap_updator/New(minimap, ztarget)
+	..()
+	src.minimap = minimap
+	src.ztarget = ztarget
+
+/**
+ * Adds an atom we want to track with blips to the subsystem
+ * Arguments:
+ * * target: atom we want to track
+ * * zlevel: zlevel we want thhis atom to be tracked for
+ * * hud_flags: tracked HUDs we want this atom to be displayed on
+ * * iconstate: iconstate for the blip we want to be used for this tracked atom
+ * * icon: icon file we want to use for this blip, 'icons/UI_icons/map_blips.dmi' by efault
+ */
+/datum/controller/subsystem/minimaps/proc/add_marker(atom/target, zlevel, hud_flags = NONE, iconstate, icon = 'icons/UI_icons/map_blips.dmi')
+	if(!isatom(target) || !zlevel || !hud_flags || !iconstate || !icon)
+		CRASH("Invalid marker added to subsystem")
+	if(!initialized)
```

Comment:
> 
> 
> Seems weird that this proc can be called before the system is initialized, i guess it's because of the low priority?

stuff outside of the SS calls add_marker on initialize() so depending of initialize order or something it might happen

Reply:
^^